src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { AppMenuActionId, AppMenuActionType, AppMenuRegistry } from '@kbn/discover-utils';
+import { DATA_QUALITY_LOCATOR_ID, DataQualityLocatorParams } from '@kbn/deeplinks-observability';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { isOfQueryType } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { AppMenuExtensionParams } from '../../../..';
+import type { RootProfileProvider } from '../../../../profiles';
+import { ProfileProviderServices } from '../../../profile_provider_services';
+
+export const createGetAppMenu =
+  (services: ProfileProviderServices): RootProfileProvider['profile']['getAppMenu'] =>
+  (prev) =>
+  (params) => {
+    const prevValue = prev(params);
+
+    return {
+      appMenuRegistry: (registry) => {
+        // Register custom link actions
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.datasets', {
+          defaultMessage: 'Data sets',
+        }),
+        testId: 'discoverAppMenuDatasetQualityLink',
+        onClick: ({ onFinishAction }) => {
```

Comment:
Nit: Is it possible to make this an actual link? Feels like it would be better if users could also open in a new tab if they wanted to.

Reply:
I initially tried to use it as a normal link, however it seems the shared navigation top nav is not playing well with links (I couldn't find any link usage in Discover and Lens), and the changes to support this seem out of the scope for this work.

Also, pre-computing the URL when registering the menu entry cause de-syncing the parameters from the updates, which result in navigating to the dataset quality without the applied parameters. This is also why it needs to get the current parameters on click.

On the other hand, IMO the shared aggregate top nav introduces a lot of logic and constraints which might be not necessary for the usages it has, which is only in Discover and Lens top nav... Having a simpler implementation of the top nav directly in Discover (implementing links as any other app is doing), would probably simplify this a lot and give us more flexibility on the app menu entries. Just food for thoughts, is surely not scoped for this work to change.